PhotoApp/MainViewController/Other/FriendsVewComtroller/UserViewController/UserViewController.swift

```diff
@@ -9,130 +9,233 @@
+                        if (snapshot.value as? String) == "1" {
+                            self.status = .youSubsribeForHim
+                            self.ref.child(self.user.id).child("subscribes").child(userID).removeAllObservers()
+                            //self.ref.child(userID).child("subscribes").child(self.user.id).removeAllObservers()
                             self.setupObservers()
                             self.setupDesign()
                         }
                     }
+                    self.ref.child(userID).child("subscribes").child(user.id).observe(.value) { (snapshot) in
+                                           if (snapshot.value as? String) == "1" {
+                                               self.status = .heSubscribeForYou
+                                               self.ref.child(userID).child("subscribes").child(self.user.id).removeAllObservers()
+                                               // self.ref.child(self.user.id).child("subscribes").child(userID).removeAllObservers()
+                                               self.setupObservers()
+                                               self.setupDesign()
+                                           }
+                                       }
                 default:
                     print("kffk")
                 }
-            
+
         }
     }
-    func setupDesign(){
+    func setupDesign() {
+
         switch status {
-        case .Friend:
-            self.usernameLabel.text = self.user.username
-            self.avatarImageView.loadImageWhithoutUser(idString: "\(user.id)/\(user.id)")
+        case .friend:
+
             self.friendBtn.backgroundColor = UIColor.init(named: "subfribackground")
             self.friendBtn.setTitleColor(UIColor.init(named: "subfrietintcolor"), for: .normal)
             self.friendBtn.setTitle("In Friends", for: .normal)
-            
-            
-            
+
         case .youSubsribeForHim:
-            self.usernameLabel.text = self.user.username
-            self.avatarImageView.loadImageWhithoutUser(idString: "\(user.id)/\(user.id)")
+           // self.usernameLabel.text = self.user.username
+            //self.avatarImageView.loadImageWhithoutUser(idString: "\(user.id)/\(user.id)")
             self.friendBtn.backgroundColor = UIColor.init(named: "subfribackground-1")
             self.friendBtn.setTitleColor(UIColor.init(named: "subfrietintcolor"), for: .normal)
             self.friendBtn.setTitle("You subscribed", for: .normal)
-            
+
         case .heSubscribeForYou:
-            self.usernameLabel.text = self.user.username
-            self.avatarImageView.loadImageWhithoutUser(idString: "\(user.id)/\(user.id)")
+           // self.usernameLabel.text = self.user.username
+           // self.avatarImageView.loadImageWhithoutUser(idString: "\(user.id)/\(user.id)")
```

Comment:
@Kasiama comments? Are they still needed? Also check previous and next lines

Reply:
u are right, done